### PR TITLE
Add better support for ZSH and adds the regex dependency

### PR DESCRIPTION
**Pull request by anemovva**
This gets rid of 0 showing up as the most common command by removing the zsh prefix with regex. I also had to add the regex dependency to the project.

### CODE REVIEW

File: src/hist_file.rs [hunk 1]
```diff
@@ -46,6 +47,11 @@ pub fn parse_contents(contents: String, args: &Args) -> HashMap<String, usize> {
         });
         only_prefix.push_str("\n");
     }
+
+    let reg = Regex::new(r": \d\d\d\d\d\d\d\d\d\d:\d;").unwrap();
```
Comment (nthnd) [inline src/hist_file.rs]: Could it be possible to let users provide this regex as an arguement? That way, we could cover similar edge cases in other shells as well

Comment (anemovva) [reply src/hist_file.rs]: Alright I'll work on adding that. It might take a while though, as I'm pretty new to rust.

Comment (nthnd) [reply src/hist_file.rs]: That's alright. I appreciate any help

**Comment by anemovva** [comment]
I added the option to change the regex string as an argument by adding `--regexp="regular expression string here"`. I still have it defaulting to zsh, as I feel like that is probably the most common shell other than bash, but I can change that if you want.

**Comment by anemovva** [comment]
I see that theres issue #17 that I might also be able work on, would that be helpful?

**Comment by anemovva** [comment]
I got bored so I went ahead and addressed #17 , let me know if you don't want it and I can just undo it.

**Review by nthnd — APPROVED** [review]
LGTM

**Pull request merged**